weblate/api/views.py

```diff
@@ -1154,6 +1154,24 @@ def units(self, request, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK,)
 
+    @action(detail=True, methods=["delete"], url_path="units/(?P<unit_id>[^/.]+)")
+    def delete_units(self, request, pk, unit_id):
+        obj = self.get_object()
+        if not request.user.has_perm("screenshot.edit", obj.component):
+            raise PermissionDenied()
+
+        try:
+            source_string = obj.component.source_translation.unit_set.get(
+                pk=int(unit_id)
+            )
+        except (Unit.DoesNotExist, ValueError) as error:
+            return Response(
+                data={"result": "Unsuccessful", "detail": force_str(error)},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
Can you please cover this by unit tests?

Reply:
Added.